src/AutoPilotPlugins/PX4/AirframeComponentController.cc

```diff
@@ -127,7 +127,7 @@ void AirframeComponentController::_waitParamWriteSignal(QVariant value)
         // Now that both params have made it to the vehicle we can reboot it. All these signals are flying
         // around on the main thread, so we need to allow the stack to unwind back to the event loop before
         // we reboot.
-        QTimer::singleShot(100, this, &AirframeComponentController::_rebootAfterStackUnwind);
+        QTimer::singleShot(800, this, &AirframeComponentController::_rebootAfterStackUnwind);
```

Comment:
Was this not working? Reason I wonder is that the amount of time here shouldn't really matter. It's just to unwind the stack. If it wasn't working then it's likely an indicator that this whole new mechanism I added isn't really working correctly.